RELEASE.md

```diff
@@ -11,11 +11,17 @@ Please follow the established format:
 
 ## Major features and improvements
 
-<!-- Add release notes for the upcoming release here -->
+- Complete backend rewrite to be more modular and maintainable using FastAPI. (#432)
```

Comment:
@limdauto I recall the new backend requires an update to Kedro 0.17.4 - it would be worth to mention it in the release notes ( and ideally in the contribution docs?)